pages/movies.tsx

```diff
@@ -0,0 +1,223 @@
+  const [totalPagesState, setTotalPages] = useState(totalPages);
+
+  const fetchMovies = async (
+    query: string,
+    page: number,
+    type: string | null,
+    year: string | null
+  ) => {
+    const apiKey = 'b3e9b3df';
+    let url = `http://www.omdbapi.com/?apikey=${apiKey}&s=${query}&page=${page}&per_page=10`;
+
+    if (type) {
+      url += `&type=${type}`;
+    }
+
+    if (year) {
+      url += `&y=${year}`;
+    }
+
+    const response = await fetch(url);
+    const data = await response.json();
+
+    if (data.Search) {
+      setMovies(data.Search);
+      setSearchedMovie(data.Search.length > 0 ? data.Search[0] : null);
+      setTotalPages(Math.ceil(data.totalResults / 10));
+    } else {
+      setMovies([]);
+      setSearchedMovie(null);
+      setTotalPages(0);
+    }
+  };
+
+  const handleSearch = async () => {
+    await fetchMovies(searchTermState, 1, selectedTypeState, selectedYearState);
+  };
+
+  const handlePageChange = (page: number) => {
+    setCurrentPage(page);
+    fetchMovies(searchTermState, page, selectedTypeState, selectedYearState);
+  };
+
+  useEffect(() => {
+    const fetchMoviesAndSetState = async () => {
+      await fetchMovies(searchTermState, currentPageState, selectedTypeState, selectedYearState);
+    };
+
+    fetchMoviesAndSetState();
+  }, [searchTermState, currentPageState, selectedTypeState]);
+
+  return (
+    <Layout>
+      <div className="flex flex-col items-center h-screen">
+        <div className="w-full max-w-screen-lg flex-grow">
+          <h1 className="text-3xl font-bold mb-4">Movie List</h1>
+
+          <div className="flex items-center mb-4">
+            <input
+              type="text"
+              placeholder="Search for a movie"
+              value={searchTermState}
+              onChange={(e) => setSearchTerm(e.target.value)}
+              className="border p-2 mr-2 flex-grow bg-gray-700 text-white placeholder-gray-400 focus:outline-none"
+            />
+            <select
+              value={selectedTypeState || ''}
+              onChange={(e) => setSelectedType(e.target.value)}
+              className="border p-2 bg-gray-700 text-white focus:outline-none"
+            >
+              <option value="">All</option>
+              <option value="movie">Movie</option>
+              <option value="series">Series</option>
+              <option value="episode">Episode</option>
+            </select>
+            <input
+              type="text"
+              placeholder="Year"
+              value={selectedYearState || ''}
+              onChange={(e) => setSelectedYear(e.target.value)}
+              className="border p-2 bg-gray-700 text-white focus:outline-none"
+            />
+            <button
+              onClick={handleSearch}
+              className="bg-blue-500 text-white px-4 py-2 rounded focus:outline-none"
+            >
+              Search
+            </button>
```

Comment:
**issue (blocking)**: please replace all the component with the Shadcn's component (`<Input />`, `<Select />`, `<Button />`, etc).